src/commands/createFunction/FunctionListStep.ts

```diff
@@ -64,7 +66,12 @@ export class FunctionListStep extends AzureWizardPromptStep<IFunctionWizardConte
                 promptSteps: [new PythonLocationStep(this._functionSettings)]
             };
         } else {
-            return await FunctionSubWizard.createSubWizard(context, this._functionSettings);
+            const needsDurableStorageSetup: boolean = !!context.functionTemplate && durableUtils.requiresDurableStorage(context.functionTemplate.id, context.language) && !context.hasDurableStorage;
```

Comment:
I think it's cleaner to move all the other conditions into the `requiresDurableStorage` function. If you just pass the entire context, you can do all of these checks in the function.